tests/migrations/test_autodetector.py

```diff
@@ -32,14 +32,227 @@ def deconstruct(self):
         return (self.__module__ + "." + self.__class__.__name__, self.args, self.kwargs)
 
 
-class AutodetectorTests(TestCase):
+class BaseAutodetectorTests(TestCase):
+    author_empty = ModelState(
+        "testapp", "Author", [("id", models.AutoField(primary_key=True))]
+    )
+    book = ModelState(
+        "otherapp",
+        "Book",
+        [
+            ("id", models.AutoField(primary_key=True)),
+            ("author", models.ForeignKey("testapp.Author", models.CASCADE)),
+            ("title", models.CharField(max_length=200)),
+        ],
+    )
+    book_with_no_author = ModelState(
+        "otherapp",
+        "Book",
+        [
+            ("id", models.AutoField(primary_key=True)),
+            ("title", models.CharField(max_length=200)),
+        ],
+    )
+    author_with_book = ModelState(
+        "testapp",
+        "Author",
+        [
+            ("id", models.AutoField(primary_key=True)),
+            ("name", models.CharField(max_length=200)),
+            ("book", models.ForeignKey("otherapp.Book", models.CASCADE)),
+        ],
+    )
+    book_indexes = ModelState(
+        "otherapp",
+        "Book",
+        [
+            ("id", models.AutoField(primary_key=True)),
+            ("author", models.ForeignKey("testapp.Author", models.CASCADE)),
+            ("title", models.CharField(max_length=200)),
+        ],
+        {
+            "indexes": [
+                models.Index(fields=["author", "title"], name="book_title_author_idx")
+            ],
+        },
+    )
```

Comment:
I would keep these states in `AutodetectorTests`. You can use them in `AutodetectorIndexTogetherTests` as a class properties, e.g.:
```python
        self.get_changes(
            [AutodetectorTests.book, AutodetectorTests.author_with_book],
            [...],
        )
```